kernel/src/scan/mod.rs

```diff
@@ -244,21 +261,27 @@ impl Scan {
             };
 
             let read_results =
-                parquet_handler.read_parquet_files(&[meta], read_schema.clone(), None)?;
+                parquet_handler.read_parquet_files(&[meta], self.physical_schema.clone(), None)?;
 
-            let read_expression = if have_partition_cols {
+            let read_expression = if self.have_partition_cols
+                || self.column_mapping_mode != ColumnMappingMode::None
+            {
```

Comment:
I think we can revert this now? CMM no longer affects the logic inside this block?

Reply:
As you noted in https://github.com/delta-incubator/delta-kernel-rs/pull/205/files#r1610218802, we still need this